geoh5py/objects/object_base.py

```diff
@@ -117,17 +124,24 @@ def add_data(
         """
         data_objects = []
         for name, attr in data.items():
-            assert isinstance(attr, dict), (
-                f"Given value to data {name} should of type {dict}. "
-                f"Type {type(attr)} given instead."
-            )
+            if not isinstance(attr, dict):
+                raise TypeError(
+                    f"Given value to data {name} should of type {dict}. "
+                    f"Type {type(attr)} given instead."
+                )
+
             attr["name"] = name
-            self.validate_data_association(attr)
-            entity_type = DataType.validate_data_type(self.workspace, attr)
+            attr, validate_property_group = self.validate_association(
```

Comment:
would be nice to raise a warning if association=CELLS, and len(values) != len cells, etc.

Reply:
This function is just to auto-detected the association based on the length, not the other way around.